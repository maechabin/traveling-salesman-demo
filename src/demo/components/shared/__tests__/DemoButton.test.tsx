import React from 'react';
import { shallow } from 'enzyme';

import DemoButton from '../DemoButton';

describe('<Demobutton />', () => {
  it('propsから受け取った値を適切に表示すること', () => {
    // setup
    const callbackSpy = jest.fn();
    const props = {
      callback: callbackSpy,
      isDisabled: true,
      classname: 'AAA',
      label: 'BBB',
    };
    const wrapper = shallow(<DemoButton {...props} />);

    // verify
    expect(wrapper.childAt(0).text()).toBe(props.label);
    expect(wrapper.find('button').props().disabled).toBe(true);
    expect(wrapper.find('button').props().className).toBe(props.classname);
    expect(wrapper).toMatchSnapshot();
  });

  it('ボタンをクリックした時にprops.callbackが呼び出されること', () => {
    // setup
    const callbackSpy = jest.fn();
    const props = {
      callback: callbackSpy,
      isDisabled: true,
      classname: 'AAA',
      label: 'BBB',
    };
    const wrapper = shallow(<DemoButton {...props} />);

    // exercise
    wrapper.find('button').simulate('click');

    // verify
    expect(callbackSpy).toHaveBeenCalled();
  });
});
