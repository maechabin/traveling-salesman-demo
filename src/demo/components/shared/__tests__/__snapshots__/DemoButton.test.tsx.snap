// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Demobutton /> propsから受け取った値を適切に表示すること 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DemoButton
    callback={[MockFunction]}
    classname="AAA"
    isDisabled={true}
    label="BBB"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "BBB",
      "className": "AAA",
      "disabled": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "BBB",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "BBB",
        "className": "AAA",
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "BBB",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
